src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const combinedStates = [];
+
+  for (const action of actions) {
```

Comment:
You can destructure action right here:
    _**for (const { type, extraData, keysToRemove } of actions)**_ 
    Doing so, you won't have to reach the properties later using dot notation (line 14, 16), and use destructuring in line 20.